In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.api as sm
import seaborn as sns
from sklearn.model_selection import train_test_split
from scipy import stats

In [2]:
pd.options.display.float_format = '{:.2f}'.format

# Dataset loading

In [3]:
data_path = r"C:\Users\erica\Documents\Mestrado\Projetos\Advanced_DS\Ames_Housing_Project\data\raw\AmesHousing.csv"
df = pd.read_csv(data_path)
pd.set_option('display.max_columns', None)
df.head()


,Order,PID,MS SubClass,MS Zoning,Lot Frontage,Lot Area,Street,Alley,Lot Shape,Land Contour,Utilities,Lot Config,Land Slope,Neighborhood,Condition 1,Condition 2,Bldg Type,House Style,Overall Qual,Overall Cond,Year Built,Year Remod/Add,Roof Style,Roof Matl,Exterior 1st,Exterior 2nd,Mas Vnr Type,Mas Vnr Area,Exter Qual,Exter Cond,Foundation,Bsmt Qual,Bsmt Cond,Bsmt Exposure,BsmtFin Type 1,BsmtFin SF 1,BsmtFin Type 2,BsmtFin SF 2,Bsmt Unf SF,Total Bsmt SF,Heating,Heating QC,Central Air,Electrical,1st Flr SF,2nd Flr SF,Low Qual Fin SF,Gr Liv Area,Bsmt Full Bath,Bsmt Half Bath,Full Bath,Half Bath,Bedroom AbvGr,Kitchen AbvGr,Kitchen Qual,TotRms AbvGrd,Functional,Fireplaces,Fireplace Qu,Garage Type,Garage Yr Blt,Garage Finish,Garage Cars,Garage Area,Garage Qual,Garage Cond,Paved Drive,Wood Deck SF,Open Porch SF,Enclosed Porch,3Ssn Porch,Screen Porch,Pool Area,Pool QC,Fence,Misc Feature,Misc Val,Mo Sold,Yr Sold,Sale Type,Sale Condition,SalePrice
0,1,526301100,20,RL,141.00,31770,Pave,NaN,IR1,Lvl,AllPub,Corner,Gtl,NAmes,Norm,Norm,1Fam,1Story,6,5,1960,1960,Hip,CompShg,BrkFace,Plywood,Stone,112.00,TA,TA,CBlock,TA,Gd,Gd,BLQ,639.00,Unf,0.00,441.00,1080.00,GasA,Fa,Y,SBrkr,1656,0,0,1656,1.00,0.00,1,0,3,1,TA,7,Typ,2,Gd,Attchd,1960.00,Fin,2.00,528.00,TA,TA,P,210,62,0,0,0,0,NaN,NaN,NaN,0,5,2010,WD,Normal,215000
1,2,526350040,20,RH,80.00,11622,Pave,NaN,Reg,Lvl,AllPub,Inside,Gtl,NAmes,Feedr,Norm,1Fam,1Story,5,6,1961,1961,Gable,CompShg,VinylSd,VinylSd,NaN,0.00,TA,TA,CBlock,TA,TA,No,Rec,468.00,LwQ,144.00,270.00,882.00,GasA,TA,Y,SBrkr,896,0,0,896,0.00,0.00,1,0,2,1,TA,5,Typ,0,NaN,Attchd,1961.00,Unf,1.00,730.00,TA,TA,Y,140,0,0,0,120,0,NaN,MnPrv,NaN,0,6,2010,WD,Normal,105000
2,3,526351010,20,RL,81.00,14267,Pave,NaN,IR1,Lvl,AllPub,Corner,Gtl,NAmes,Norm,Norm,1Fam,1Story,6,6,1958,1958,Hip,CompShg,Wd Sdng,Wd Sdng,BrkFace,108.00,TA,TA,CBlock,TA,TA,No,ALQ,923.00,Unf,0.00,406.00,1329.00,GasA,TA,Y,SBrkr,1329,0,0,1329,0.00,0.00,1,1,3,1,Gd,6,Typ,0,NaN,Attchd,1958.00,Unf,1.00,312.00,TA,TA,Y,393,36,0,0,0,0,NaN,NaN,Gar2,12500,6,2010,WD,Normal,172000
3,4,526353030,20,RL,93.00,11160,Pave,NaN,Reg,Lvl,AllPub,Corner,Gtl,NAmes,Norm,Norm,1Fam,1Story,7,5,1968,1968,Hip,CompShg,BrkFace,BrkFace,NaN,0.00,Gd,TA,CBlock,TA,TA,No,ALQ,1065.00,Unf,0.00,1045.00,2110.00,GasA,Ex,Y,SBrkr,2110,0,0,2110,1.00,0.00,2,1,3,1,Ex,8,Typ,2,TA,Attchd,1968.00,Fin,2.00,522.00,TA,TA,Y,0,0,0,0,0,0,NaN,NaN,NaN,0,4,2010,WD,Normal,244000
4,5,527105010,60,RL,74.00,13830,Pave,NaN,IR1,Lvl,AllPub,Inside,Gtl,Gilbert,Norm,Norm,1Fam,2Story,5,5,1997,1998,Gable,CompShg,VinylSd,VinylSd,NaN,0.00,TA,TA,PConc,Gd,TA,No,GLQ,791.00,Unf,0.00,137.00,928.00,GasA,Gd,Y,SBrkr,928,701,0,1629,0.00,0.00,2,1,3,1,TA,6,Typ,1,TA,Attchd,1997.00,Fin,2.00,482.00,TA,TA,Y,212,34,0,0,0,0,NaN,MnPrv,NaN,0,3,2010,WD,Normal,189900


## Descriptive statistics

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2930 entries, 0 to 2929
Data columns (total 82 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Order            2930 non-null   int64  
 1   PID              2930 non-null   int64  
 2   MS SubClass      2930 non-null   int64  
 3   MS Zoning        2930 non-null   str    
 4   Lot Frontage     2440 non-null   float64
 5   Lot Area         2930 non-null   int64  
 6   Street           2930 non-null   str    
 7   Alley            198 non-null    str    
 8   Lot Shape        2930 non-null   str    
 9   Land Contour     2930 non-null   str    
 10  Utilities        2930 non-null   str    
 11  Lot Config       2930 non-null   str    
 12  Land Slope       2930 non-null   str    
 13  Neighborhood     2930 non-null   str    
 14  Condition 1      2930 non-null   str    
 15  Condition 2      2930 non-null   str    
 16  Bldg Type        2930 non-null   str    
 17  House Style      2930 non

In [5]:
df.describe()

,Order,PID,MS SubClass,Lot Frontage,Lot Area,Overall Qual,Overall Cond,Year Built,Year Remod/Add,Mas Vnr Area,BsmtFin SF 1,BsmtFin SF 2,Bsmt Unf SF,Total Bsmt SF,1st Flr SF,2nd Flr SF,Low Qual Fin SF,Gr Liv Area,Bsmt Full Bath,Bsmt Half Bath,Full Bath,Half Bath,Bedroom AbvGr,Kitchen AbvGr,TotRms AbvGrd,Fireplaces,Garage Yr Blt,Garage Cars,Garage Area,Wood Deck SF,Open Porch SF,Enclosed Porch,3Ssn Porch,Screen Porch,Pool Area,Misc Val,Mo Sold,Yr Sold,SalePrice
count,2930.00,2930.00,2930.00,2440.00,2930.00,2930.00,2930.00,2930.00,2930.00,2907.00,2929.00,2929.00,2929.00,2929.00,2930.00,2930.00,2930.00,2930.00,2928.00,2928.00,2930.00,2930.00,2930.00,2930.00,2930.00,2930.00,2771.00,2929.00,2929.00,2930.00,2930.00,2930.00,2930.00,2930.00,2930.00,2930.00,2930.00,2930.00,2930.00
mean,1465.50,714464496.99,57.39,69.22,10147.92,6.09,5.56,1971.36,1984.27,101.90,442.63,49.72,559.26,1051.61,1159.56,335.46,4.68,1499.69,0.43,0.06,1.57,0.38,2.85,1.04,6.44,0.60,1978.13,1.77,472.82,93.75,47.53,23.01,2.59,16.00,2.24,50.64,6.22,2007.79,180796.06
std,845.96,188730844.65,42.64,23.37,7880.02,1.41,1.11,30.25,20.86,179.11,455.59,169.17,439.49,440.62,391.89,428.40,46.31,505.51,0.52,0.25,0.55,0.50,0.83,0.21,1.57,0.65,25.53,0.76,215.05,126.36,67.48,64.14,25.14,56.09,35.60,566.34,2.71,1.32,79886.69
min,1.00,526301100.00,20.00,21.00,1300.00,1.00,1.00,1872.00,1950.00,0.00,0.00,0.00,0.00,0.00,334.00,0.00,0.00,334.00,0.00,0.00,0.00,0.00,0.00,0.00,2.00,0.00,1895.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,1.00,2006.00,12789.00
25%,733.25,528477022.50,20.00,58.00,7440.25,5.00,5.00,1954.00,1965.00,0.00,0.00,0.00,219.00,793.00,876.25,0.00,0.00,1126.00,0.00,0.00,1.00,0.00,2.00,1.00,5.00,0.00,1960.00,1.00,320.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,4.00,2007.00,129500.00
50%,1465.50,535453620.00,50.00,68.00,9436.50,6.00,5.00,1973.00,1993.00,0.00,370.00,0.00,466.00,990.00,1084.00,0.00,0.00,1442.00,0.00,0.00,2.00,0.00,3.00,1.00,6.00,1.00,1979.00,2.00,480.00,0.00,27.00,0.00,0.00,0.00,0.00,0.00,6.00,2008.00,160000.00
75%,2197.75,907181097.50,70.00,80.00,11555.25,7.00,6.00,2001.00,2004.00,164.00,734.00,0.00,802.00,1302.00,1384.00,703.75,0.00,1742.75,1.00,0.00,2.00,1.00,3.00,1.00,7.00,1.00,2002.00,2.00,576.00,168.00,70.00,0.00,0.00,0.00,0.00,0.00,8.00,2009.00,213500.00
max,2930.00,1007100110.00,190.00,313.00,215245.00,10.00,9.00,2010.00,2010.00,1600.00,5644.00,1526.00,2336.00,6110.00,5095.00,2065.00,1064.00,5642.00,3.00,2.00,4.00,2.00,8.00,3.00,15.00,4.00,2207.00,5.00,1488.00,1424.00,742.00,1012.00,508.00,576.00,800.00,17000.00,12.00,2010.00,755000.00


In [6]:
print(f"Quantidade de linhas: {df.shape[0]}")
print(f"Quantidade de colunas: {df.shape[1]}")

Quantidade de linhas: 2930
Quantidade de colunas: 82


### Variáveis numéricas

In [7]:
print(df.select_dtypes(include='number').columns.tolist())

['Order', 'PID', 'MS SubClass', 'Lot Frontage', 'Lot Area', 'Overall Qual', 'Overall Cond', 'Year Built', 'Year Remod/Add', 'Mas Vnr Area', 'BsmtFin SF 1', 'BsmtFin SF 2', 'Bsmt Unf SF', 'Total Bsmt SF', '1st Flr SF', '2nd Flr SF', 'Low Qual Fin SF', 'Gr Liv Area', 'Bsmt Full Bath', 'Bsmt Half Bath', 'Full Bath', 'Half Bath', 'Bedroom AbvGr', 'Kitchen AbvGr', 'TotRms AbvGrd', 'Fireplaces', 'Garage Yr Blt', 'Garage Cars', 'Garage Area', 'Wood Deck SF', 'Open Porch SF', 'Enclosed Porch', '3Ssn Porch', 'Screen Porch', 'Pool Area', 'Misc Val', 'Mo Sold', 'Yr Sold', 'SalePrice']


In [8]:
numerical_cols = df.select_dtypes(include='number').drop(columns=['Order','PID', 'MS SubClass','Overall Qual', 'Overall Cond'])
numerical_cols.columns

Index(['Lot Frontage', 'Lot Area', 'Year Built', 'Year Remod/Add',
       'Mas Vnr Area', 'BsmtFin SF 1', 'BsmtFin SF 2', 'Bsmt Unf SF',
       'Total Bsmt SF', '1st Flr SF', '2nd Flr SF', 'Low Qual Fin SF',
       'Gr Liv Area', 'Bsmt Full Bath', 'Bsmt Half Bath', 'Full Bath',
       'Half Bath', 'Bedroom AbvGr', 'Kitchen AbvGr', 'TotRms AbvGrd',
       'Fireplaces', 'Garage Yr Blt', 'Garage Cars', 'Garage Area',
       'Wood Deck SF', 'Open Porch SF', 'Enclosed Porch', '3Ssn Porch',
       'Screen Porch', 'Pool Area', 'Misc Val', 'Mo Sold', 'Yr Sold',
       'SalePrice'],
      dtype='str')

### Variáveis Categóricas

In [9]:
categorical_cols = df.select_dtypes(include=['object', 'string']).columns
extra_cols = pd.Index(['MS SubClass', 'Overall Qual', 'Overall Cond'])
categorical_cols = categorical_cols.append(extra_cols)
categorical_cols

Index(['MS Zoning', 'Street', 'Alley', 'Lot Shape', 'Land Contour',
       'Utilities', 'Lot Config', 'Land Slope', 'Neighborhood', 'Condition 1',
       'Condition 2', 'Bldg Type', 'House Style', 'Roof Style', 'Roof Matl',
       'Exterior 1st', 'Exterior 2nd', 'Mas Vnr Type', 'Exter Qual',
       'Exter Cond', 'Foundation', 'Bsmt Qual', 'Bsmt Cond', 'Bsmt Exposure',
       'BsmtFin Type 1', 'BsmtFin Type 2', 'Heating', 'Heating QC',
       'Central Air', 'Electrical', 'Kitchen Qual', 'Functional',
       'Fireplace Qu', 'Garage Type', 'Garage Finish', 'Garage Qual',
       'Garage Cond', 'Paved Drive', 'Pool QC', 'Fence', 'Misc Feature',
       'Sale Type', 'Sale Condition', 'MS SubClass', 'Overall Qual',
       'Overall Cond'],
      dtype='str')

## Análise univariada

### Variáveis numéricas

In [ ]:
baixa_variabilidade = []
alta_variabilidade = []
moderada_variabilidade = []
skewness_high = []
skewness_moderate = []
skewness_low = []
kurtosis_high = []
kurtosis_moderate = [] 
kurtosis_low = []


for col in numerical_cols:
    print(f"===== Coluna {col} =====\n")

    print(f"{df[col].describe()}\n")

    cv = df[col].std() / df[col].mean()

    if cv < 0.3:
        print(f"Coeficiente de Variação: CV = {cv:.2f} » Baixa variabilidade - valores concentrados em torno da média")
        baixa_variabilidade.append(col)
    if cv >= 0.7:
        print(f"Coeficiente de Variação: CV = {cv:.2f} » Alta variabilidade - valores mais dispersos em relação à média")
        alta_variabilidade.append(col)
    if 0.3 <= cv < 0.7:
        print(f"Coeficiente de Variação: CV = {cv:.2f} » Variabilidade moderada - pode indicar assimetria dos dados e/ou presença de outliers")
        moderada_variabilidade.append(col)

    iqr = df[col].quantile(0.75) - df[col].quantile(0.25)
    print(f"Range: {df[col].max() - df[col].min()}")

    print(f"IQR: {iqr}")
    print(f"Limite inferior: {df[col].quantile(0.25) - 1.5 * iqr}")
    print(f"Limite superior: {df[col].quantile(0.75) + 1.5 * iqr}")
    print(f"Quantidade de outliers: {((df[col] < (df[col].quantile(0.25) - 1.5 * iqr)) | (df[col] > (df[col].quantile(0.75) + 1.5 * iqr))).sum()}")

    skewness = df[col].skew()
    kurtosis = df[col].kurtosis()

    if skewness < 0.5:
        print(f"Assimetria: {skewness:.2f} » Approximately symmetric")
        skewness_low.append(col)
    elif 0.5 <= skewness < 1:
        print(f"Assimetria: {skewness:.2f} » Moderately skewed")
        skewness_moderate.append(col)
    else:
        print(f"Assimetria: {skewness:.2f} » Highly skewed")
        skewness_high.append(col)

    if kurtosis == 0:
        print(f"Curtose: {kurtosis:.2f} » Similar to normal distribution")
        kurtosis_low.append(col)
    elif 0 < kurtosis < 3:
        print(f"Curtose: {kurtosis:.2f} » Moderately heavy tails")
        kurtosis_moderate.append(col)
    else:
        print(f"Curtose: {kurtosis:.2f} » Heavy tails and more extreme values")
        kurtosis_high.append(col)

    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(6, 3))
    ax1.boxplot(df[col].dropna())
    ax1.set_title(f"Boxplot da coluna {col}")
    ax2.hist(df[col], bins=30, edgecolor='black', )
    ax2.set_title(f"Histograma da coluna {col}")
    fig.suptitle(f"Análise da coluna {col}")
    plt.tight_layout()
    plt.show()

    print("=" * 80)

In [ ]:
# criar uma função que verifique se cada coluna está presente em cada uma das listas de variabilidade,
# assimetria e curtose, para facilitar a análise posterior
for column in numerical_cols:
    print(f"\n===== Analisando a coluna: {column} ====")
    
    if column in baixa_variabilidade:
        print("Variabilidade: Baixa")
    elif column in moderada_variabilidade:
        print("Variabilidade: Moderada")
    elif column in alta_variabilidade:
        print("Variabilidade: Alta")
    else:
        print("Variabilidade: Não classificada")

    if column in skewness_low:
        print("Assimetria: Baixa (Approximately symmetric)")
    elif column in skewness_moderate:
        print("Assimetria: Moderada (Moderately skewed)")
    elif column in skewness_high:
        print("Assimetria: Alta (Highly skewed)")
    else:
        print("Assimetria: Não classificada")

    if column in kurtosis_low:
        print("Curtose: Baixa (Similar to normal distribution)")
    elif column in kurtosis_moderate:
        print("Curtose: Moderada (Moderately heavy tails)")
    elif column in kurtosis_high:
        print("Curtose: Alta (Heavy tails and more extreme values)")
    else:
        print("Curtose: Não classificada")

In [12]:
skewness_values = []
for col in numerical_cols.columns:
    skewness = df[col].skew()
    skewness_values.append(skewness)

skewness_df = pd.DataFrame({
    'Coluna': numerical_cols.columns,
    'Assimetria': skewness_values
})
skewness_df.groupby(pd.cut(skewness_df['Assimetria'], bins=[-float('inf'), 0.5, 1, float('inf')], labels=['Aproximadamente Simétrica', 'Moderadamente Assimétrica', 'Altamente Assimétrica'])).size()

Assimetria
Aproximadamente Simétrica     9
Moderadamente Assimétrica     6
Altamente Assimétrica        19
dtype: int64

In [13]:
cols_assimetria_alta = skewness_df[skewness_df['Assimetria'] >= 1].sort_values(by='Assimetria', ascending=False)[['Coluna']].values
cols_assimetria_alta

array([['Misc Val'],
       ['Pool Area'],
       ['Lot Area'],
       ['Low Qual Fin SF'],
       ['3Ssn Porch'],
       ['Kitchen AbvGr'],
       ['BsmtFin SF 2'],
       ['Enclosed Porch'],
       ['Screen Porch'],
       ['Bsmt Half Bath'],
       ['Mas Vnr Area'],
       ['Open Porch SF'],
       ['Wood Deck SF'],
       ['SalePrice'],
       ['Lot Frontage'],
       ['1st Flr SF'],
       ['BsmtFin SF 1'],
       ['Gr Liv Area'],
       ['Total Bsmt SF']], dtype=object)

### Plano de Engenharia de Variáveis e Transformações

| Variável | Transformação | Observação |
| :--- | :--- | :--- |
| **Lot Frontage** | log(x+1) | Imputar mediana por Neighborhood antes |
| **Lot Area** | log(x+1) | Verificar outliers acima de 4000ft² antes |
| **Year Built** | house_age = 2025 - Year Built | Substituir por idade da casa |
| **Year Remod/Add** | years_since_remod = 2025 - Year Remod/Add | Substituir por anos desde reforma |
| **Mas Vnr Area** | log(x+1) | — |
| **BsmtFin SF 1** | log(x+1) | — |
| **BsmtFin SF 2** | log(x+1) | — |
| **Bsmt Unf SF** | log(x+1) | — |
| **Total Bsmt SF** | log(x+1) | — |
| **1st Flr SF** | log(x+1) | — |
| **2nd Flr SF** | log(x+1) | — |
| **Low Qual Fin SF** | log(x+1) | — |
| **Gr Liv Area** | log(x+1) | Verificar outliers acima de 4000ft² antes |
| **Bsmt Full Bath** | nenhuma | — |
| **Bsmt Half Bath** | nenhuma | — |
| **Full Bath** | nenhuma | — |
| **Half Bath** | nenhuma | — |
| **Bedroom AbvGr** | nenhuma | — |
| **Kitchen AbvGr** | nenhuma | — |
| **TotRms AbvGrd** | nenhuma | — |
| **Fireplaces** | nenhuma | Criar has_fireplace (0 ou 1) |
| **Garage Yr Blt** | imput Year remod/add | Corrigir ano de construção da garagem 2207 |
| **Garage Cars** | nenhuma | — |
| **Garage Area** | nenhuma | — |
| **Wood Deck SF** | log(x+1) | Criar has_deck (0 ou 1) |
| **Open Porch SF** | log(x+1) | — |
| **Enclosed Porch** | log(x+1) | — |
| **3Ssn Porch** | log(x+1) | — |
| **Screen Porch** | log(x+1) | — |
| **Pool Area** | descartar | Substituir por has_pool (0 ou 1) |
| **Misc Val** |

### Variáveis Categoricas

In [14]:
cols_baixa_variabilidade = []


for col in categorical_cols:
    valores_unicos = df[col].nunique()
    mais_frequente = df[col].mode().iloc[0]
    #frequencia_mais_frequente = df[col].value_counts().iloc[0]
    porcentagem_mais_frequente = df[col].value_counts(normalize=True, dropna=False).iloc[0] * 100
    #menos_frequente = df[col].value_counts().index[-1]
    #frequencia_menos_frequente = df[col].value_counts().iloc[-1]
    porcentagem_menos_frequente = df[col].value_counts(normalize=True, dropna=False).iloc[-1] * 100
    categorias_raras = []
    
    for cat in df[col]:
        try:
            if df[col].value_counts(normalize=True, dropna=False)[cat] < 0.05:
                categorias_raras.append(cat)
        except KeyError:
            pass

    print(f"\n===== {col} =====\n")
    #print(f"Valores únicos: {valores_unicos}")
    #print(f"Valor mais frequente: {mais_frequente}")
    #print(f"Frequência do valor mais frequente: {frequencia_mais_frequente}")
    #print(f"Porcentagem do valor mais frequente: {porcentagem_mais_frequente:.2f}%")
    print(f"Contagem de cada categoria:\n{df[col].value_counts(dropna=False)}\n")
    #print(f"Valor menos frequente: {menos_frequente}")
    #print(f"Frequência do valor menos frequente: {frequencia_menos_frequente}")
    #print(f"Porcentagem do valor menos frequente: {porcentagem_menos_frequente:.2f}%")
    print(f"Categorias raras (menos de 5% dos dados): {set(categorias_raras)}")

    if porcentagem_mais_frequente > 90:
        #print(f"\n===== {col} =====\n")
        cols_baixa_variabilidade.append(col)
        print(f"Coluna pode ter baixo poder discriminatório (a categoria -{mais_frequente}- representa {porcentagem_mais_frequente:.2f}% dos dados)")

    if valores_unicos >= 10 and valores_unicos < 30:
        print(f" tem cardinalidade moderada ({valores_unicos} valores únicos)")
    if valores_unicos >= 30 and valores_unicos < 50:
        print(f" tem cardinalidade alta ({valores_unicos} valores únicos)")
    if valores_unicos >= 50:
        print(f"tem cardinalidade muito alta (Não é um bom preditor para modelos lineares)")
    if valores_unicos < 10:
        print(f"tem cardinalidade baixa ({valores_unicos} valores únicos)")
    
    


===== MS Zoning =====

Contagem de cada categoria:
MS Zoning
RL         2273
RM          462
FV          139
RH           27
C (all)      25
I (all)       2
A (agr)       2
Name: count, dtype: int64

Categorias raras (menos de 5% dos dados): {'C (all)', 'I (all)', 'A (agr)', 'FV', 'RH'}
tem cardinalidade baixa (7 valores únicos)

===== Street =====

Contagem de cada categoria:
Street
Pave    2918
Grvl      12
Name: count, dtype: int64

Categorias raras (menos de 5% dos dados): {'Grvl'}
Coluna pode ter baixo poder discriminatório (a categoria -Pave- representa 99.59% dos dados)
tem cardinalidade baixa (2 valores únicos)

===== Alley =====

Contagem de cada categoria:
Alley
NaN     2732
Grvl     120
Pave      78
Name: count, dtype: int64

Categorias raras (menos de 5% dos dados): {'Pave', 'Grvl'}
Coluna pode ter baixo poder discriminatório (a categoria -Grvl- representa 93.24% dos dados)
tem cardinalidade baixa (2 valores únicos)

===== Lot Shape =====

Contagem de cada categoria:
Lot S

### Planejamento de Tratamento de Variáveis Categóricas

| Variável | Status | Ação | Detalhes |
| :--- | :--- | :--- | :--- |
| **MS Zoning** | Pendente | Agrupar categorias raras | A (agr), C (all), FV, RH, I (all) → Outros |
| **Street** | Pendente | Excluir | Baixo valor preditivo — 99.59% Pave |
| **Lot Shape** | Pendente | Agrupar | IR1, IR2, IR3 → Irregular / Reg → Regular |
| **Land Contour** | Pendente | Excluir | Baixo valor preditivo |
| **Utilities** | Pendente | Excluir | Baixo valor preditivo — 99.90% AllPub |
| **Lot Config** | Pendente | Agrupar | Inside → Inside / demais → Outros |
| **Land Slope** | Pendente | Excluir | Baixo valor preditivo — 95.19% Gtl |
| **Neighborhood** | Pendente | Agrupar por faixa de preço médio | Baixo / Médio / Alto — definir faixas após análise |
| **Condition 1** | Pendente | Agrupar categorias raras | PosA, RRNn, PosN, Artery, RRNe, RRAn, RRAe → Outros |
| **Condition 2** | Pendente | Excluir | Baixo valor preditivo — 98.98% Norm |
| **Bldg Type** | Pendente | Agrupar categorias raras | Duplex, Twnhs, 2fmCon → Outros |
| **House Style** | Pendente | Agrupar categorias raras | 1.5Unf, SFoyer, SLvl, 2.5Unf, 2.5Fin → Outros |
| **Roof Style** | Pendente | Agrupar categorias raras | Mansard, Flat, Shed, Gambrel → Outros |
| **Roof Matl** | Pendente | Excluir | Baixo valor preditivo — 98.53% CompShg |
| **Exterior 1st** | Pendente | Agrupar categorias raras | Stucco, AsbShng, WdShing, BrkFace, Stone, etc → Outros |
| **Exterior 2nd** | Pendente | Excluir | Redundante com Exterior 1st |
| **Mas Vnr Type** | Pendente | Agrupar categorias raras, nos 7 valores com area contruida inputar moda | CBlock, BrkCmn → Outros |
| **Exter Qual** | Pendente | Encoding ordinal | Po=1, Fa=2, TA=3, Gd=4, Ex=5 |
| **Exter Cond** | Pendente | Encoding ordinal | Po=1, Fa=2, TA=3, Gd=4, Ex=5 |
| **Foundation** | Pendente | Agrupar categorias raras | Stone, Wood, Slab → Outros |
| **Bsmt Qual** | Pendente | Encoding ordinal | None=0, Po=1, Fa=2, TA=3, Gd=4, Ex=5 |
| **Bsmt Cond** | Pendente | Excluir | Redundante com Bsmt Qual |
| **Bsmt Exposure** | Pendente | Agrupar | Av, Gd, Mn → Yes / No → No / None → None |
| **BsmtFin Type 1** | Aguarda correlação | Manter ou excluir | Decidir após análise de correlação com SalePrice |
| **BsmtFin Type 2** | Pendente | Agrupar | Rec, LwQ, BLQ, ALQ, GLQ → Com acabamento / Unf → Unf |
| **Heating** | Pendente | Excluir | Baixo valor preditivo — 98.46% GasA |
| **Heating QC** | Pendente